src/server/routes/entity/process-unified-form.ts

```diff
@@ -0,0 +1,241 @@
+import * as achievement from '../../helpers/achievement';
+import * as handler from '../../helpers/handler';
+import type {Request as $Request, Response as $Response} from 'express';
+import {
+	fetchEntitiesForRelationships, fetchOrCreateMainEntity, getChangedProps,
+	getNextRelationshipSets, indexAutoCreatedEditionGroup, saveEntitiesAndFinishRevision
+} from './entity';
+import type {
+	EntityTypeString
+} from 'bookbrainz-data/lib/func/types';
+import _ from 'lodash';
+import {_bulkIndexEntities} from '../../../common/helpers/search';
+import {transformNewForm as authorTransform} from './author';
+import {transformNewForm as editionGroupTransform} from './edition-group';
+import {transformNewForm as editionTransform} from './edition';
+import log from 'log';
+import {transformNewForm as publisherTransform} from './publisher';
+import {transformNewForm as seriesTransform} from './series';
+import {transformNewForm as workTransform} from './work';
+
+
+type PassportRequest = $Request & {user: any, session: any};
+const transformFunctions = {
+	author: authorTransform,
+	edition: editionTransform,
+	editionGroup: editionGroupTransform,
+	publisher: publisherTransform,
+	series: seriesTransform,
+	work: workTransform
+};
+const additionalEntityProps = {
+	author: [
+		'typeId', 'genderId', 'beginAreaId', 'beginDate', 'endDate', 'ended',
+		'endAreaId'
+	],
+	edition: [
+		'editionGroupBbid', 'width', 'height', 'depth', 'weight', 'pages',
+		'formatId', 'statusId'
+	],
+	editionGroup: 'typeid',
+	publisher: ['typeId', 'areaId', 'beginDate', 'endDate', 'ended'],
+	series: ['entityType', 'orderingTypeId'],
+	work: 'typeId'
+
+};
+
+export async function processAchievement(orm, editorId, entityJSON) {
+	const {revisionId} = entityJSON;
+	try {
+		const achievementUnlock = await achievement.processEdit(orm, editorId, revisionId);
+		if (achievementUnlock.alert) {
+			entityJSON.alert = achievementUnlock.alert;
+		}
+		return entityJSON;
+	}
+	catch (err) {
+		return Promise.reject(new Error(err));
+	}
+}
+
+export function transformForm(body:Record<string, any>):Record<string, any> {
+	const modifiedForm = {};
+	for (const keyIndex in body) {
+		if (Object.prototype.hasOwnProperty.call(body, keyIndex)) {
+			const currentForm = body[keyIndex];
+			const transformedForm = transformFunctions[_.lowerFirst(currentForm.type)](currentForm);
```

Comment:
The tests are failing because of this line; we should probably play it safe in any case and either catch any possible error or first store the transformfunction in a variable, then check if the funcion exists and is actually a function (isFunction lodash utility) and only then call it.